packages/styled-components/src/utils/stylis.js

```diff
@@ -48,7 +48,7 @@ export default function createStylisInstance({
    */
   const selfReferenceReplacementPlugin = element => {
     if (element.type === RULESET && element.value.includes('&')) {
-      element.props[0] = element.props[0].replace(_selectorRegexp, selfReferenceReplacer);
+      element.props = element.props.map(prop => prop.replace(_selectorRegexp, selfReferenceReplacer))
```

Comment:
Could you use a for loop for this instead of ES5 `map`? Generally there's a perf penalty for those methods since it has to generate a new scope.

Reply:
Sure.